### DISCUSSION TITLE
Convert PNG Image received via a web service call to base64 string

### DISCUSSION THREAD

**Discussion by Gabenny** [SalesforceCommerceCloud/sfcc-ci#420] [Team Posts]
Hello,
When calling a web service I'm receive a png format file that I want to convert to a base64 format string.
Can someone help me please figure out how to do this knowing that png definition is in response.object.client.text and it comes in a weird format.
Thank you in advance for your reply
Best regards

**Comment by pravngaur** [comment]
Hi,

First you need to setup the right content type in request, i think if you
are working with images, image/png should work for you. This should give
you png file not text.

Second, to convert base you can use something like
https://stackoverflow.com/questions/24523532/how-do-i-convert-an-image-to-a-base64-encoded-data-url-in-sails-js-or-generally
.

This should work, let me know how it goes.

**Comment by Gabenny** [comment]
Hello,
Thank you for your reply.
Actually I need to develop a controller that receives a base64 string and uploads it to the IMPEX as a png file
I must not use forms.
do you have any clue on how to do this?
Thank you in advance for your reply

**Comment by zlatinz** [comment]
Hi @Gabenny ,
There is a quota `api.dw.io.File.createNewFile()` with Storefront Limit: 0.
See https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/quota/html/API_Quotas.html#category_I/O%20and%20Network

Which means that you cannot create files from storefront.
What you can do instead is capture the string in custom object and have job that creates the file. (e.g. using `StringUtils.decodeBase64` method) https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/scriptapi/html/api/class_dw_util_StringUtils.html#dw_util_StringUtils_decodeBase64_String_String_DetailAnchor
You may need to play with the encoding parameter. Usually `ISO-8859-1` as encoding does the trick, you can also try `UTF-8`
